src/client/mod.rs

```diff
@@ -132,6 +133,19 @@ where C: Connect,
     pub fn into_compat(self) -> compat::CompatClient<C, B> {
         self::compat::client(self)
     }
+
+    fn enforce_version(&self, version: &mut HttpVersion) {
+        match (self.version, *version) {
+            (Ver::Http1, HttpVersion::Http10) |
+            (Ver::Http1, HttpVersion::Http11) => {},
+            (Ver::Http1, _) => {
+                *version = HttpVersion::Http11;
+            },
+            _ => {
+
+            }
+        }
```

Comment:
Just randomly stumbled upon this repo. I think there's a way to simplify this:
```Rust
match (self.version, *version) {
    (Ver::Http1, _) if *version != HttpVersion::Http10 || *version != HttpVersion::Http11 => {
        *version = HttpVersion::Http11;
    },
    _ => {},
}
```
This way you don't need the first match arm :).